src/Connections/GitlabRepo.ts

```diff
@@ -512,6 +563,29 @@ ${data.description}`;
         }
         return false;
     }
+
+    public async provisionerUpdateConfig(userId: string, config: Record<string, unknown>) {
+        const validatedConfig = validateState(config);
+        await this.as.botClient.sendStateEvent(this.roomId, GitLabRepoConnection.CanonicalEventType, this.stateKey, 
+            {
+                ...validatedConfig,
+            }
+        );
+    }
+
+
+    public async onRemove() {
+        log.info(`Removing ${this.toString()} for ${this.roomId}`);
+        // Do a sanity check that the event exists.
+        try {
+            await this.as.botClient.getRoomStateEvent(this.roomId, GitLabRepoConnection.CanonicalEventType, this.stateKey);
+            await this.as.botClient.sendStateEvent(this.roomId, GitLabRepoConnection.CanonicalEventType, this.stateKey, { disabled: true });
+        } catch (ex) {
+            await this.as.botClient.getRoomStateEvent(this.roomId, GitLabRepoConnection.LegacyCanonicalEventType, this.stateKey);
+            await this.as.botClient.sendStateEvent(this.roomId, GitLabRepoConnection.LegacyCanonicalEventType, this.stateKey, { disabled: true });
+        }
+        // TODO: Clean up webhooks
```

Comment:
:eyes: What webhooks need to be cleaned up here?
Is this required for a merge?

Reply:
We set up GitLab webhooks when setting up the connection, but we also need to remove them when they are no longer in use.

The trouble here is that we don't have context on whether other rooms are also using the same webhooks, so we don't want to vape them in case they are still used. I'll write some code for this.